libcxx/test/std/atomics/atomics.lockfree/isalwayslockfree.pass.cpp

```diff
@@ -21,7 +21,6 @@
 template <typename T>
 void checkAlwaysLockFree() {
   if (std::atomic<T>::is_always_lock_free) {
-    LIBCPP_ASSERT(sizeof(std::atomic<T>) == sizeof(T)); // technically not required, but libc++ does it that way
```

Comment:
I assume the FloatArr3 case is what causes this?
Under what architecture? I haven't been able to reproduce the size difference in godbolt.

Reply:
Yes, exactly, `FloatArr3` showcases this under `arm64-apple-macos` (and probably any `arm64` target). I looked at the assembly and we do not use a lock for `FloatArr3` on arm64 so it looks like `is_lock_free` was really lying.